lms/djangoapps/bookmarks/views.py

```diff
@@ -107,7 +107,7 @@ def get_queryset(self):
             log.error("Invalid course id '{course_id}'")
             return []
 
-        results_queryset = Bookmark.objects.filter(course_key=course_key, user=self.request.user).order_by('-created')
+        results_queryset = Bookmark.objects.filter(course_key=course_key, user=self.request.user).select_related('block_cache').order_by('-created')
```

Comment:
Please add a test asserting the number of queries needed to load bookmarks (and also the number of queries used when creating new bookmarks or updating existing bookmarks).